go/vt/vttablet/tabletmanager/tm_state.go

```diff
@@ -261,18 +266,47 @@ func (ts *tmState) updateLocked(ctx context.Context) {
 	}
 
 	if ts.isShardServing[ts.tablet.Type] {
-		ts.isInSRVKeyspace = true
-		statsIsInSRVKeyspace.Set(1)
+		ts.isInSrvKeyspace = true
+		statsIsInSrvKeyspace.Set(1)
 	} else {
-		ts.isInSRVKeyspace = false
-		statsIsInSRVKeyspace.Set(0)
+		ts.isInSrvKeyspace = false
+		statsIsInSrvKeyspace.Set(0)
 	}
 
 	// Open TabletServer last so that it advertises serving after all other services are up.
 	if reason == "" {
 		if err := ts.tm.QueryServiceControl.SetServingType(ts.tablet.Type, terTime, true, ""); err != nil {
 			log.Errorf("Cannot start query service: %v", err)
 		}
+
+		ts.populateLocalMetadataLocked()
+	}
+}
+
+func (ts *tmState) populateLocalMetadataLocked() {
+	if ts.tm.MetadataManager == nil {
+		return
+	}
+
+	if ts.isOpening && !*initPopulateMetadata {
+		return
+	}
+
+	localMetadata := ts.tm.getLocalMetadataValues(ts.tablet.Type)
+	dbName := topoproto.TabletDbName(ts.tablet)
+
+	if !ts.hasCreatedMetadataTables {
```

Comment:
It is not obvious to me why this check is needed. It seems that with the initial populatemetadata in tm_init this check should't be required.

Reply:
Hah, I agree. It took me a while to convince myself I had this correct:

* In tm_init, during `handleRestore`:
    * if `-restore_from_backup` was set, then we populate (and create) the metadata tables as a side effect
    * if `-init_populate_metadata` was set, then we populate (and create) the metadata tables explicitly
    * if neither flag was set, the metadata tables do not get created

This check is there to cover that third case, where neither flag was passed to the `vttablet` binary. This was discussed a bit in https://github.com/vitessio/vitess/pull/5783, in particular:

> Note that we have chosen not to change the behavior of metadata_tables if `-init_populate_metadata` is false in the presence of `-restore_from_backup` because doing so would break backwards compatibility.

Maybe now is the right time to revisit that, and break that older behavior to simplify this.

Reply:
Oh I see. That makes sense. Maybe we should add a comment about this. Because it's not an obvious insight .

One more nit - I would set the following inside PopulateMetadataTables:
```
		ts.hasCreatedMetadataTables = true
```